src/python/pants/task/simple_codegen_task.py

```diff
@@ -293,6 +332,21 @@ def _inject_synthetic_target(
 
     return synthetic_target
 
+  def _supports_exports(self, target_type):
+    return hasattr(target_type, 'export_specs')
+
+  def _original_export_specs(self, target):
+    collected_original_exports = getattr(target, 'export_specs', None)
```

Comment:
I believe I left this comment before, but: this should be able to use `Target.exports`, no?

Reply:
Sure. 

I think the reason I didn't is to avoid the extra dependency traversals in target.py's `_get_synthetic_target` since we're doing them at use later anyway. The other issue is that it won't include the codegen target in the export list, which I haven't thought through the implications of.

I can work around both of those by calling it with an object that has codegen_types set to an empty collection.

The cache entries there should be removed when the targets are invalidated. I could write a test that verifies that.

Reply:
We'll still need the check on the synthetic type though. I could move that method to Target I suppose.